tests/functional/aliases/test_aliases.py

```diff
@@ -0,0 +1,339 @@
+    def models(self):
+        return {
+            "schema.yml": models__schema_yml,
+            "foo_alias.sql": models__foo_alias_sql,
+            "alias_in_project.sql": models__alias_in_project_sql,
+            "alias_in_project_with_override.sql": models__alias_in_project_with_override_sql,
+            "ref_foo_alias.sql": models__ref_foo_alias_sql,
+        }
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"cast.sql": macros__cast_sql, "expect_value.sql": macros__expect_value_sql}
+
+    def test_alias_model_name(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 4
+        run_dbt(["test"])
+
+
+class TestAliasErrors:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "macro-paths": ["macros"],
+        }
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"cast.sql": macros__cast_sql, "expect_value.sql": macros__expect_value_sql}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "model_b.sql": models_dupe__model_b_sql,
+            "model_a.sql": models_dupe__model_a_sql,
+            "README.md": models_dupe__README_md,
+        }
+
+    def test_alias_dupe_thorews_exeption(self, project):
+        message = ".*identical database representation.*"
+        with pytest.raises(Exception) as exc:
+            assert message in exc
+            run_dbt(["run"])
+
+
+class TestSameAliasDifferentSchemas:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "macro-paths": ["macros"],
+        }
+
+    @pytest.fixture(scope="class")
+    def macros(self):
+        return {"cast.sql": macros__cast_sql, "expect_value.sql": macros__expect_value_sql}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": models_dupe_custom_schema__schema_yml,
+            "model_c.sql": models_dupe_custom_schema__model_c_sql,
+            "model_b.sql": models_dupe_custom_schema__model_b_sql,
+            "model_a.sql": models_dupe_custom_schema__model_a_sql,
+            "README.md": models_dupe_custom_schema__README_md,
+        }
+
+    def test_same_alias_succeeds_in_different_schemas(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+        res = run_dbt(["test"])
+        assert len(res) > 0
+
+
+class TestSameAliasDifferentDatabases:
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "config-version": 2,
+            "macro-paths": ["macros"],
+            "models": {
+                "test": {
+                    "alias": "duped_alias",
+                    "model_b": {
+                        # how to tap into a alternate database here? or is something like dbt_profile_target method required
```

Comment:
I think you'll need to add a call to the project fixture's create_test_schema(schemaname="someAlternateDB") as that looks like the equivalent functionality